### ISSUE TITLE
Investigate special linker script instead of a dynamic library for libstdc++.so

### ISSUE THREAD

**Issue by probonopd** [AppImage/AppImageKit#1063]
Currently, we recommend applications to be built on the oldest still-supported LTS release of Ubuntu. However, some applications need newer C++ compilers, which means that developers usually pull in a PPA that results in binaries being compiled in a way such that they don't work with the original libstdc++ that originally came with the oldest still-supported LTS release of Ubuntu.

Apparently CentOS devtoolset offers a solution for this issue.

We should understand how this magic in CentOS devtoolset works, and how it can be used with Ubuntu LTS...

https://stackoverflow.com/questions/52609269/c-project-compiled-with-modern-compiler-but-linked-against-outdated-libstdc/52611144#52611144

> The `devtoolset-6-gcc-c++` package provides a custom version of GCC that uses a special linker script instead of a dynamic library for libstdc++.so. That means the binaries it produces **do not depend on the newer libstdc++.so.6** and can be run on other CentOS machines that don't have devtoolset installed (i.e. they only have the older `libstdc++` library from GCC 4.8).

We should then document it in docs.appimage.org...

Reference:
https://github.com/wheybags/glibc_version_header/issues/20

**Comment by TheAssassin** [comment]
Well, you'll still have to build your tool with that script then. Now that CentOS 6 has been deprecated, it might be easier to just have people use CentOS 7 (released 2014) with their vanilla GCC (which fully supports C++11 and I think C++14) or maybe `devtoolset-8` or `devtoolset-9`.

**Comment by probonopd** [comment]
> Now that CentOS 6 has been deprecated

Oh, interesting to hear.

For RHEL 6, 
* Maintenance Support or Maintenance Support 2 ended 30. November 2020.
* Extended life cycle support (ELS) add-on ends 30. Juni 2024.

So I guess it's fine to recommend CentOS 7 for new implementations, but to keep CentOS 6 for https://github.com/AppImage/AppImageKit for the forseeable future. Do you agree?

While in general I am not a big fan of Red Hat for a multitude of reasons, I do have to say "hats off" for their devtoolset becuase it shows that _someone_ at Red Hat seems to think about (and understand) the concept of backward compatibility. I wish they would talk more to the Gnome team ;-)

**Comment by TheAssassin** [comment]
> So I guess it's fine to recommend CentOS 7 for new implementations, but to keep CentOS 6 for https://github.com/AppImage/AppImageKit for the forseeable future. Do you agree?

No, I don't agree. You always said the baseline was CentOS 6. It's dead now. And it's super old. Whatever "ELS" means isn't really visible anyway, who knows if they really update anything. And we don't benefit from that anyway without getting hands on RHEL.

Just switch to CentOS 7.
